public/script.js

```diff
@@ -4637,7 +4639,7 @@ export async function sendMessageAsUser(messageText, messageBias, insertAt = nul
     await populateFileAttachment(message);
     statMesProcess(message, 'user', characters, this_chid, '');
 
-    if (typeof insertAt === 'number' && insertAt >= 0 && insertAt <= chat.length) {
+    if (insertAtDepth) {
```

Comment:
This would fail if insertAtDepth is 0. Cause it will instead push the message at the end.

Reply:
You are right and now I am fucking embarrassed. Blame it in JS deciding that value `0` equals false.